client/src/main/java/com/vesoft/nebula/entry/AbstractNebulaCallback.java

```diff
@@ -0,0 +1,60 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+package com.vesoft.nebula.entry;
+
+import com.facebook.thrift.TBase;
+import com.facebook.thrift.TException;
+import com.facebook.thrift.async.AsyncMethodCallback;
+import com.facebook.thrift.async.TAsyncMethodCall;
+import com.google.common.base.Optional;
+
+import java.util.concurrent.atomic.AtomicBoolean;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public abstract class AbstractNebulaCallback implements AsyncMethodCallback {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AbstractNebulaCallback.class);
+
+    protected TBase result;
+
+    protected AtomicBoolean isReady = new AtomicBoolean(false);
+
+    public Optional<?> getResult() throws InterruptedException {
+        while (!checkReady()) {
+            Thread.sleep(100);
+        }
+        return result == null ? Optional.absent() : Optional.of(result);
```

Comment:
We can make sure the result is not empty, so maybe return `Optional.of(result)` is OK .

Reply:
👍